### PR TITLE
Update restrictKey to a function using filterWithKey 

### PR DESCRIPTION
**Pull request by leopoulson**
When running `cabal build` on the project, cabal errors, giving us the error message:

    src/Gossip/Caas.hs:38:53: error:
        Not in scope: ‘IntMap.restrictKeys’
        Module ‘Data.IntMap’ does not export ‘restrictKeys’.

It seems that restrictKeys has been deprecated. However, we can use the helpful equality from [here](https://hackage.haskell.org/package/containers-0.5.10.2/docs/Data-Map-Strict.html#v:restrictKeys) to provide an easy way around this problem.

When this update is applied, all tests are passed when running `stack clean; stack test --coverage`.

### CODE REVIEW

**Comment by m4lvin** [comment]
Thanks for pointing this out!

I am confused whether you are using cabal or stack? I think even without the fix "stack clean; stack test --coverage" should work because stack will use the ghc version and packages specified by the resolver in `stack.yaml`. In contrast, `cabal build` will use whatever version of ghc is globally installed on your system. (But then, I almost never use cabal ...)

For GHC 8.0.1 (with `resolver:lts-7.24` in `stack.yaml`) the version of containers I get is containers-0.5.7.1 and that does indeed not export restrictKeys. But in later versions of `containers` the function is there. Though it was not deprecated, but rather the opposite, it was only introduced later. (Also note that it is not in `IntMap` itself but reexported from the `Lazy` or `Internal` module, I think.)

It also seems that `restrictKeys` is equal to the `filterWithKey` expression, but there is a difference in runtime complexity. So I would prefer to keep using the shorter and maybe more efficient restrictKeys. Supporting older versions of GHC or libraries is not a priority for me... I would rather make it explicit in the package description that a version of containers >= 0.5.8 is needed. This should then trigger an earlier warning if `cabal build` uses an older version of ghc.

tl/dr: I really like stack :wink:

**Comment by leopoulson** [comment]
Ah I see now; the difference came from me mixing up build tools! Running `stack build` on the project on my machine does not cause the message. I didn't know that about the two tools & I thank you for letting me know! 

Of course you're right; there's no need to use a slower function to pander to people using older version of GHC or libraries. Apologies for jumping the gun on this and not checking thoroughly.
